mapflow/mapflow.py

```diff
@@ -3140,6 +3179,40 @@ def update_processing(self):
         dialog.setup(processing)
         dialog.deleteLater()
 
+    def check_project_sharing(self):
```

Comment:
I would split it into 2 functions:
- get_project_sharing(project) which parses thee project sharing data and retrieves the role and owner data
- set_ui_for_user_role(user_role, owner_email), which should be moved to the mainDialog

Reply:
I am not sure we need the second one.
To move `helpers.generate_plugin_header` we would need to pass all 5 arguments, so it doesn't matter if we call for it or for some dlg function that calls for headers (?)
Now (after changes) all other lines seem to be in place (definition of self.user_role, self.current_project, etc.)